# In this notebook, basic exploration of the given dataset is being done. Since the dataset given initially had a duplicate gene_id probe (total 18466 rows), I used Phantasus to collapse common gene_id based on maximum median probe condition and I finally ended up with 18465 rows as there was only single gene_id which had two values. I downloaded the dataset from the phantasus platform for further analysis  

# Since I'm using python programming language for the analysis, I used the cmapPy module for reading the GTC file to parse the whole file in form of pandas dataframe

In [1]:
#Loading necessary modules
from cmapPy.pandasGEXpress.parse import parse
import numpy as np
import pandas as pd

In [2]:
#loading dataset with redundant gene ID's collapsed
df=parse('./PAAD_collapsed.gct')

In [6]:
#Shape of the dataset
df.data_df.shape

(18465, 183)

## Taking a look at column metadata of the 183 example, we find information rich columns containing general information of the experiments along with information about the histological_type information of the different experiments

In [10]:
#looking at Column metadata to find out 
df.col_metadata_df

chd                      participant_id          sample_type  mRNAseq_cluster  \
cid                                                                             
aab1-Primary solid Tumor           aab1  Primary solid Tumor              1.0   
aab4-Primary solid Tumor           aab4  Primary solid Tumor              2.0   
aab6-Primary solid Tumor           aab6  Primary solid Tumor              3.0   
aab8-Primary solid Tumor           aab8  Primary solid Tumor              1.0   
aab9-Primary solid Tumor           aab9  Primary solid Tumor              4.0   
aaba-Primary solid Tumor           aaba  Primary solid Tumor              3.0   
aabe-Primary solid Tumor           aabe  Primary solid Tumor              3.0   
aabf-Primary solid Tumor           aabf  Primary solid Tumor              2.0   
aabh-Primary solid Tumor           aabh  Primary solid Tumor              2.0   
aabi-Primary solid Tumor           aabi  Primary solid Tumor              3.0   
aabk-Primary solid Tumor           aabk  Primary solid Tumor              2.0   
aabo-Primary solid Tumor           aabo  Primary solid Tumor              3.0   
aabp-Primary solid Tumor           aabp  Primary solid Tumor              5.0   
aabr-Primary solid Tumor           aabr  Primary solid Tumor              4.0   
aabt-Primary solid Tumor           aabt  Primary solid Tumor              5.0   
aabu-Primary solid Tumor           aabu  Primary solid Tumor              3.0   
aabv-Primary solid Tumor           aabv  Primary solid Tumor              4.0   
aaqa-Primary solid Tumor           aaqa  Primary solid Tumor              1.0   
aaqe-Primary solid Tumor           aaqe  Primary solid Tumor              3.0   
aaqi-Primary solid Tumor           aaqi  Primary solid Tumor              4.0   
aaqj-Primary solid Tumor           aaqj  Primary solid Tumor              1.0   
aaql-Primary solid Tumor           aaql  Primary solid Tumor              2.0   
aaqm-Primary solid Tumor           aaqm  Primary solid Tumor              2.0   
a9i5-Primary solid Tumor           a9i5  Primary solid Tumor              2.0   
a9i7-Primary solid Tumor           a9i7  Primary solid Tumor              5.0   
a9i9-Primary solid Tumor           a9i9  Primary solid Tumor              4.0   
a9ib-Primary solid Tumor           a9ib  Primary solid Tumor              3.0   
a9ic-Primary solid Tumor           a9ic  Primary solid Tumor              3.0   
a9ih-Primary solid Tumor           a9ih  Primary solid Tumor              2.0   
a9ij-Primary solid Tumor           a9ij  Primary solid Tumor              2.0   
...                                 ...                  ...              ...   
a8f3-Primary solid Tumor           a8f3  Primary solid Tumor              4.0   
a9q5-Primary solid Tumor           a9q5  Primary solid Tumor              4.0   
a5n4-Primary solid Tumor           a5n4  Primary solid Tumor              3.0   
a75w-Primary solid Tumor           a75w  Primary solid Tumor              1.0   
a5re-Primary solid Tumor           a5re  Primary solid Tumor              1.0   
a5qy-Primary solid Tumor           a5qy  Primary solid Tumor              5.0   
aa2a-Primary solid Tumor           aa2a  Primary solid Tumor              1.0   
a7b8-Primary solid Tumor           a7b8  Primary solid Tumor              1.0   
aa9m-Primary solid Tumor           aa9m  Primary solid Tumor              2.0   
aaas-Primary solid Tumor           aaas  Primary solid Tumor              4.0   
a8rm-Primary solid Tumor           a8rm  Primary solid Tumor              1.0   
a8ro-Primary solid Tumor           a8ro  Primary solid Tumor              3.0   
a8rp-Primary solid Tumor           a8rp  Primary solid Tumor              1.0   
a774-Primary solid Tumor           a774  Primary solid Tumor              4.0   
a776-Primary solid Tumor           a776  Primary solid Tumor              1.0   
a779-Primary solid Tumor           a779  Primary solid Tumor              1.0   
a77e-Primary solid Tumor       

In [13]:
#looking at histological Column 
df.col_metadata_df.histological_type

cid
aab1-Primary solid Tumor               pancreas-adenocarcinoma-other subtype
aab4-Primary solid Tumor               pancreas-adenocarcinoma-other subtype
aab6-Primary solid Tumor                 pancreas-adenocarcinoma ductal type
aab8-Primary solid Tumor                 pancreas-adenocarcinoma ductal type
aab9-Primary solid Tumor                 pancreas-adenocarcinoma ductal type
aaba-Primary solid Tumor                 pancreas-adenocarcinoma ductal type
aabe-Primary solid Tumor                 pancreas-adenocarcinoma ductal type
aabf-Primary solid Tumor                 pancreas-adenocarcinoma ductal type
aabh-Primary solid Tumor               pancreas-adenocarcinoma-other subtype
aabi-Primary solid Tumor               pancreas-adenocarcinoma-other subtype
aabk-Primary solid Tumor                 pancreas-adenocarcinoma ductal type
aabo-Primary solid Tumor                 pancreas-adenocarcinoma ductal type
aabp-Primary solid Tumor                 pancreas-undifferentiated carci

In [14]:
#counting histological_type counts
df.col_metadata_df.histological_type.value_counts()

pancreas-adenocarcinoma ductal type                 150
pancreas-adenocarcinoma-other subtype                27
pancreas-colloid (mucinous non-cystic) carcinoma      4
pancreas-undifferentiated carcinoma                   1
Name: histological_type, dtype: int64

## As we can see, we have four kind of histological values along with their value counts. Our task at hand is to find out and filter Exocrine tumour samples from the Neuroendocrine Tumour samples. Looking at the Histological_type values, it is not certain at first about which sampels are really Exocrine and which samples are Neuroendocrine. 

## Now taking a look at the row_metadata, we find out about gene_id information and the kind of values that it holds.

In [17]:
#now looking at the gene data
#we can see the names of all the genes
df.row_metadata_df

rhd                              Source
rid                                    
A1BG       mRNAseq_RSEM_normalized_log2
A1CF       mRNAseq_RSEM_normalized_log2
A2BP1      mRNAseq_RSEM_normalized_log2
A2LD1      mRNAseq_RSEM_normalized_log2
A2ML1      mRNAseq_RSEM_normalized_log2
A2M        mRNAseq_RSEM_normalized_log2
A4GALT     mRNAseq_RSEM_normalized_log2
A4GNT      mRNAseq_RSEM_normalized_log2
AAA1       mRNAseq_RSEM_normalized_log2
AAAS       mRNAseq_RSEM_normalized_log2
AACSL      mRNAseq_RSEM_normalized_log2
AACS       mRNAseq_RSEM_normalized_log2
AADAC      mRNAseq_RSEM_normalized_log2
AADAT      mRNAseq_RSEM_normalized_log2
AAGAB      mRNAseq_RSEM_normalized_log2
AAK1       mRNAseq_RSEM_normalized_log2
AAMP       mRNAseq_RSEM_normalized_log2
AANAT      mRNAseq_RSEM_normalized_log2
AARS2      mRNAseq_RSEM_normalized_log2
AARSD1     mRNAseq_RSEM_normalized_log2
AARS       mRNAseq_RSEM_normalized_log2
AASDHPPT   mRNAseq_RSEM_normalized_log2
AASDH      mRNAseq_RSEM_normalized_log2
AASS       mRNAseq_RSEM_normalized_log2
AATF       mRNAseq_RSEM_normalized_log2
AATK       mRNAseq_RSEM_normalized_log2
ABAT       mRNAseq_RSEM_normalized_log2
ABCA10     mRNAseq_RSEM_normalized_log2
ABCA11P    mRNAseq_RSEM_normalized_log2
ABCA12     mRNAseq_RSEM_normalized_log2
...                                 ...
ZSCAN18    mRNAseq_RSEM_normalized_log2
ZSCAN1     mRNAseq_RSEM_normalized_log2
ZSCAN20    mRNAseq_RSEM_normalized_log2
ZSCAN21    mRNAseq_RSEM_normalized_log2
ZSCAN22    mRNAseq_RSEM_normalized_log2
ZSCAN23    mRNAseq_RSEM_normalized_log2
ZSCAN29    mRNAseq_RSEM_normalized_log2
ZSCAN2     mRNAseq_RSEM_normalized_log2
ZSCAN4     mRNAseq_RSEM_normalized_log2
ZSCAN5A    mRNAseq_RSEM_normalized_log2
ZSCAN5B    mRNAseq_RSEM_normalized_log2
ZSWIM1     mRNAseq_RSEM_normalized_log2
ZSWIM3     mRNAseq_RSEM_normalized_log2
ZSWIM4     mRNAseq_RSEM_normalized_log2
ZSWIM5     mRNAseq_RSEM_normalized_log2
ZSWIM6     mRNAseq_RSEM_normalized_log2
ZSWIM7     mRNAseq_RSEM_normalized_log2
ZUFSP      mRNAseq_RSEM_normalized_log2
ZW10       mRNAseq_RSEM_normalized_log2
ZWILCH     mRNAseq_RSEM_normalized_log2
ZWINT      mRNAseq_RSEM_normalized_log2
ZXDA       mRNAseq_RSEM_normalized_log2
ZXDB       mRNAseq_RSEM_normalized_log2
ZXDC       mRNAseq_RSEM_normalized_log2
ZYG11A     mRNAseq_RSEM_normalized_log2
ZYG11B     mRNAseq_RSEM_normalized_log2
ZYX        mRNAseq_RSEM_normalized_log2
ZZEF1      mRNAseq_RSEM_normalized_log2
ZZZ3       mRNAseq_RSEM_normalized_log2
psiTPTE22  mRNAseq_RSEM_normalized_log2

[18465 rows x 1 columns]

## Finally we look at the normalized RNA_seq values for each gene_id (18465) corresponsing to different experiments (183)

In [18]:
#now looking looking at gene expression data, this is a matrix having rows as number of genes and columns as number of 
#number of experiments
df.data_df

cid        aab1-Primary solid Tumor  aab4-Primary solid Tumor  \
rid                                                             
A1BG                            6.4                       5.8   
A1CF                            4.7                       5.7   
A2BP1                          -1.0                       1.1   
A2LD1                           7.5                       6.8   
A2ML1                           6.4                       NaN   
A2M                            14.3                      14.0   
A4GALT                         10.6                      10.2   
A4GNT                           8.8                       5.6   
AAA1                            1.0                      -1.2   
AAAS                            9.4                       9.1   
AACSL                          -1.0                      -0.2   
AACS                           10.2                      10.3   
AADAC                           9.2                       5.0   
AADAT                           4.8                       6.3   
AAGAB                          10.2                      10.4   
AAK1                            9.6                       9.8   
AAMP                           11.8                      11.4   
AANAT                           NaN                       0.3   
AARS2                           8.6                       8.4   
AARSD1                          8.9                       8.7   
AARS                           11.3                      11.6   
AASDHPPT                        8.6                       9.0   
AASDH                           7.5                       7.5   
AASS                            8.1                       8.2   
AATF                           10.6                      10.2   
AATK                           10.0                       8.1   
ABAT                            9.4                       9.3   
ABCA10                          4.4                       4.9   
ABCA11P                         5.8                       5.7   
ABCA12                          1.8                       6.3   
...                             ...                       ...   
ZSCAN18                         9.1                       9.8   
ZSCAN1                          1.5                       3.1   
ZSCAN20                         6.0                       6.0   
ZSCAN21                         8.0                       7.6   
ZSCAN22                         7.3                       6.2   
ZSCAN23                         3.4                       3.7   
ZSCAN29                         8.7                       8.5   
ZSCAN2                          7.1                       7.5   
ZSCAN4                          0.5                      -1.2   
ZSCAN5A                         6.0                       5.7   
ZSCAN5B                         0.5                       0.8   
ZSWIM1                          8.6                       7.9   
ZSWIM3                          7.1                       6.2   
ZSWIM4                          8.8                       9.2   
ZSWIM5                          8.2                       6.9   
ZSWIM6                          8.8                       9.4   
ZSWIM7                          8.1                       8.1   
ZUFSP                           7.0                       7.5   
ZW10                            8.6                       8.7   
ZWILCH                          8.1                       8.1   
ZWINT                           8.6                       8.8   
ZXDA                            6.2                       5.8   
ZXDB                            9.0                       8.5   
ZXDC                            9.9                      10.0   
ZYG11A                          NaN                       NaN   
ZYG11B                          9.2                       9.3   
ZYX                            12.9                      12.4   
ZZEF1                          10.1                       9.8   
ZZZ3                            9.3

## This concludes our exploratory analysis part
# In the next step we will try to use PCA in order to look into patterns from the dataset and wether we could differentiate Exocrine tumour samples from the Neuroendocrine samples.